doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,327 @@
+
+  - Connect to building management system components from a range of vendors
+    via a range of domain specific protocols, such as BACnet and Modbus.
+  - Integrate with in-house cyber security management systems.
+
+## 2.6. Smart home
+
+Smart home use cases have monitoring and control aims which overlap with
+commercial building management. However, in the smart home there is less
+technical equipment, different protocols, and different system management
+criteria. In particular, the nodes here need to be able to:
+
+  - Connect to home networking equipment from a range of vendors via common
+    protocols for constrained devices in the home, such as BLE, 802.15.4 and
+    WiFi.
+  - Ensure the privacy of end users in an easy-to-use fashion.
+
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. In particular, this
+requires the board to:
+
+  - Support on-chip low-level wired communication.
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+In particular, this requires:
+
+  - The presence of didactic materials related to RIOT.
+  - The presence of tooling suitable to a classroom context.
+
+# 3. Design philosophies
+
+Below are the design philosophies that are typically followed by developers to
+cater for the above use cases. The sections below include descriptions of
+tradeoffs between the philosophies, and where the resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't active.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler, without demanding that users do the
+same. An idling device should conserve energy wherever possible, by default.
+For case specific power management, appropriate control should be available to
+the user.
+
+#### Small memory footprint
+
+Apart from being optimized for low memory usage, RIOT is modular so unused
+features don't use up precious RAM or flash. Almost all features are provided
+as optional modules that have to be enabled explicitly at compile time.  A
+minimal RIOT configuration starts at around <2KiB flash and <1.5KiB RAM
+(including stack space for one thread and ISRs).
+
+Starting from there, the memory usage depends on the enabled features:
+
+- Non-networked control loop / sensing applications can fit on very small MCUs
+(eg., an Atmega328P with 2KiB RAM)
+- 6lowPAN networking currently starts at ~40KiB ROM and ~10KiB RAM
+- A 6lowpan enabled CoAP server requires ~60KiB ROM and ~15KiB RAM
```

Comment:
For this amount of Flash and RAM, you can fit not only CoAP but also COSE end-to-end security with ECC in software. Worth mentioning I think. @bergzand @kaspar030 makes sense to you too?

Reply:
This sounds worth mentioning, I agree

Reply:
Has anyone got any suggestions for how to collect the metric? Eg any implementations already around involving 6LoWPAN/CoAP/COSE? I couldn't find any readily available

Reply:
Ping? Any thoughts/further opinions? Leave it out if not?

Reply:
How about you just change this sentence to:
` A 6LoWPAN enabled CoAP server and COSE end-to-end security (with ed25519 signatures in software) requires ~60KiB ROM and ~15KiB RAM`

Reply:
Changed, as discussed offline

Reply:
> A 6LoWPAN enabled CoAP server and COSE end-to-end security (with ed25519 signatures in software) requires ~60KiB ROM and ~15KiB RAM

I suggest linking to a wiki page describing the exact configuration measured. I think the mentioned 15KiB RAM usage here are not accurate.

Reply:
We could just re-state the metrics in Table III [in last year's paper](https://inet.haw-hamburg.de/papers/bghkl-rosos-18.pdf)?